src/comments.js

```diff
@@ -129,40 +140,60 @@ function attach(comments, ast, text) {
   comments.forEach(function(comment) {
     decorateComment(ast, comment, text);
 
-    var pn = comment.precedingNode;
-    var en = comment.enclosingNode;
-    var fn = comment.followingNode;
-
-    if (pn && fn) {
-      var tieCount = tiesToBreak.length;
-      if (tieCount > 0) {
-        var lastTie = tiesToBreak[tieCount - 1];
-
-        assert.strictEqual(
-          lastTie.precedingNode === comment.precedingNode,
-          lastTie.followingNode === comment.followingNode
-        );
+    const pn = comment.precedingNode;
+    const globalPn = comment.globalPrecedingNode;
+    const en = comment.enclosingNode;
+    const fn = comment.followingNode;
+    const isStartOfFile = comment.loc.start.line === 1;
+
+    if (util.hasNewline(text, locStart(comment), true) || isStartOfFile) {
```

Comment:
Hm, we use it in several places and I'd prefer to take in `{ backwards: true }` if we really want that. We're not going to remember to comment at every call site.